beater/java_attacher/discovery_rules.go

```diff
@@ -0,0 +1,81 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package javaattacher
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type discoveryRule interface {
+	include() bool
+	match(jvm *JvmDetails) bool
+	asString() string
+}
+
+type includeAllRule struct{}
+
+func (includeAllRule) match(jvm *JvmDetails) bool {
```

Comment:
**_suggestion_**: these should probably be pointer receivers to avoid copying the struct each time. See https://go.dev/tour/methods/4

(I'll only mention it here to avoid noise but it applies to all the methods in this file)